pallets/portfolio/src/lib.rs

```diff
@@ -0,0 +1,335 @@
+// This file is part of the Polymesh distribution (https://github.com/PolymathNetwork/Polymesh).
+// Copyright (c) 2020 Polymath
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, version 3.
+//
+// This program is distributed in the hope that it will be useful, but
+// WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+// General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+//! # Portfolio Module
+//!
+//! ## Overview
+//!
+//! The portfolio module provides the essential extrinsics to manage asset portfolios, public
+//! functions for integration of portfolios into other pallets, and implementations of RPC getters.
+//!
+//! ## Interface
+//!
+//! ### Dispatchable Functions
+//!
+//! - `create_portfolio`: Creates a new user portfolio.
+//! - `delete_portfolio`: Deletes an existing user portfolio.
+//! - `move_portfolio`: Moves specified amounts of assets from one portfolio to another portfolio
+//!   of the same DID.
+//! - `rename_portfolio`: Renames a user portfolio.
+//!
+//! ### Public Functions
+//!
+//! - `default_portfolio_balance`: Returns the ticker balance of the identity's default portfolio.
+//! - `user_portfolio_balance`: Returns the ticker balance of an identity's user portfolio.
+//! - `set_default_portfolio_balance`: Sets the ticker balance of the identity's default portfolio.
+//! - `rpc_get_portfolios`: An RPC function that lists all user-defined portfolio number-name pairs.
+//! - `rpc_get_portfolio_assets`: Ensures that there is no portfolio with the desired name yet.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use codec::{Decode, Encode};
+use core::result::Result;
+use frame_support::{
+    decl_error, decl_event, decl_module, decl_storage, dispatch::DispatchResult, ensure,
+    IterableStorageDoubleMap,
+};
+use frame_system::{self as system, ensure_signed};
+use polymesh_common_utilities::{identity::Trait as IdentityTrait, CommonTrait, Context};
+use polymesh_primitives::{IdentityId, PortfolioId, PortfolioName, PortfolioNumber, Ticker};
+use sp_arithmetic::traits::Saturating;
+use sp_std::{convert::TryFrom, prelude::Vec};
+
+type Identity<T> = pallet_identity::Module<T>;
+
+/// The ticker and balance of an asset to be moved from one portfolio to another.
+#[derive(Encode, Decode, Clone, Debug, Default, PartialEq, Eq, PartialOrd, Ord)]
```

Comment:
Debug only in `std` whenever possible.

Reply:
@fmiguelgarcia Does this look like a sufficient reason to still have Debug?
```
  error[E0277]: `MovePortfolioItem<<T as polymesh_common_utilities::CommonTrait>::Balance>` doesn't implement `core::fmt::Debug`
     --> /home/vladi/remote-builds/8028138931963931287/pallets/portfolio/src/lib.rs:150:1
      |
  150 | / decl_module! {
  151 | |     pub struct Module<T: Trait> for enum Call where origin: T::Origin {
  152 | |         type Error = Error<T>;
  153 | |
  ...   |
  278 | |     }
  279 | | }
      | |_^ `MovePortfolioItem<<T as polymesh_common_utilities::CommonTrait>::Balance>` cannot be formatted using `{:?}`
      |
      = help: the trait `core::fmt::Debug` is not implemented for `MovePortfolioItem<<T as polymesh_common_utilities::CommonTrait>::Balance>`
      = note: add `#[derive(Debug)]` or manually implement `core::fmt::Debug`
      = note: required because of the requirements on the impl of `core::fmt::Debug` for `sp_api_hidden_includes_decl_storage::hidden_include::dispatch::Vec<MovePortfolioItem<<T as polymesh_common_utilities::CommonTrait>::Balance>>`
      = note: required because of the requirements on the impl of `core::fmt::Debug` for `(core::option::Option<u64>, core::option::Option<u64>, sp_api_hidden_includes_decl_storage::hidden_include::dispatch::Vec<MovePortfolioItem<<T as polymesh_common_utilities::CommonTrait>::Balance>>)`
      = note: required by `core::fmt::Debug::fmt`
      = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)
```